### ISSUE TITLE
Error Converting Oracle Linux 8.3 to RHEL

### ISSUE THREAD

**Issue by shikari7** [oamg/convert2rhel#154]
After the announcement of CentOS 8.x being dropped, I used centos2ol to convert from CentOS 8.3 to Oracle Linux 8.3:

> Red Hat Enterprise Linux release 8.3 (Ootpa)

With the new RHEL announcement, I now want to convert to RHEL.  Fully updated system, using convert2rhel like so:

`convert2rhel --disable-submgr --disablerepo="*" --enablerepo BaseOS --no-rpm-va
`

And I get the following error:

```
WARNING - The tool allows rollback of any action until this point.
WARNING - By continuing all further changes on the system will need to be reverted manually by the user, if necessary.

Continue with the system conversion? [y/n]: y



[01/20/2021 22:20:33] TASK - [Convert: Import Red Hat GPG keys] *********************************

[01/20/2021 22:20:33] TASK - [Convert: Prepare kernel] ******************************************
Installing RHEL kernel ...
Unable to detect release version (use '--releasever' to specify release version)
Error: Unknown repo: 'BaseOS'
[01/20/2021 22:20:33] CRITICAL - Error occured while attempting to install the RHEL kernel
WARNING - Conversion process interrupted and manual user intervention will be necessary.

```

What am I doing wrong?    Thanks for any pointers!

**Comment by r4han85** [comment]
Did you add rhel-BaseOS repo in your directory /etc/yum.repos.d/ ?
Are files /etc/centos-release, /etc/redhat-release and /etc/os-release present ?

**Comment by bocekm** [comment]
Hi @shikari7, thanks for taking the time to file the issue.

I suppose you have the `BaseOS` repository defined in one of the repofiles in `/etc/yum.repos.d/`.
You also need to have `AppStream` repository defined and passed to convert2rhel through the --enablerepo option so then the command may look like:
`convert2rhel --disable-submgr --disablerepo="*" --enablerepo BaseOS --enablerepo AppStream --no-rpm-va `

Regarding the above, I'm quoting [the article about the conversions](https://access.redhat.com/articles/2360841):
> RHEL 8 content is distributed through two default repositories, BaseOS and AppStream. You must configure both for a successful conversion. When running the Convert2RHEL utility, make sure to enable both repositories using the --enablerepo option. For more information, see [Considerations in adopting RHEL 8](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/considerations_in_adopting_rhel_8/index#repositories_considerations-in-adopting-RHEL-8).

There's currently an issue with expanding the `$releasever` variable in a `baseurl` of repositories. Please make sure to replace the `$releasever` variable in the baseurl of the BaseOS and AppStream repositories defined in one of your `.repo` files in `/etc/yum.repos.d/`.

**Comment by bocekm** [comment]
@shikari7, with the recently released version 0.16, you can now convert the system much easier without the need to configure the above repositories manually. Read the updated article https://access.redhat.com/articles/2360841 on how to use convert2rhel with Red Hat Subscription Manager (RHSM).

Also, in the new version, we have fixed expanding the `$releasever` variable in the repository `baseurl`.
